### PR TITLE
Feature/and 174 analytics

### PR DESCRIPTION
**Pull request by vandus**

### CODE REVIEW

**Review by davidvavra — CHANGES_REQUESTED** [review]
We need approval from Jan Šicha or Jan Jirousek that this doesn't require any changes in privacy policy

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt [hunk 1]
```diff
@@ -41,6 +42,10 @@ class DashboardFragment : BaseFragment<FragmentDashboardPlusBinding, DashboardVM
 
     companion object {
         private const val SCREEN_NAME = "Dashboard"
+
+        private const val ANALYTICS_KEY_SHARE_APP = "click_share_app"
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt:46]: Can you document all Analytics events somewhere? Wiki page here on GitHub is probably the best place.

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt [hunk 2]
```diff
@@ -33,6 +33,13 @@ import javax.inject.Inject
 class MainActivity :
     BaseActivity<ActivityMainBinding, MainVM>(R.layout.activity_main, MainVM::class) {
 
+    companion object {
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt:36]: I would move this to Analytics class

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt [hunk 3]
```diff
@@ -171,16 +178,18 @@ class DashboardFragment : BaseFragment<FragmentDashboardPlusBinding, DashboardVM
...
-        exposure_notification_more_info.setOnClickListener { viewModel.showExposureDetail() }
-        exposure_notification_close.setOnClickListener {
-            viewModel.acceptExposure()
-            exposure_notification_container.hide()
-        }
 
         data_notification_close.setOnClickListener { data_notification_container.hide() }
 
+        dash_card_active.setOnClickListener {
+            Analytics.logEvent(requireContext(), ANALYTICS_KEY_PAUSE_APP)
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt:185]: Log it after the action, not before. (Analytics is secondary, if something goes wrong in analytics, it should not block the action)

**Review by davidvavra — CHANGES_REQUESTED** [review]
We now have an approval for merging. But I found few issue:

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt (hunk 2 above)
Comment (vandus) [reply app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt:36]: it's better to let the consumer control what events are sent. but they may be kept in one place, especially in a small-sized app such as this one, if you prefer.

File: app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt [hunk 4]
```diff
@@ -8,6 +8,15 @@ object Analytics {
     const val KEY_EXPORT_DOWNLOAD_STARTED = "key_export_download_started"
     const val KEY_EXPORT_DOWNLOAD_FINISHED = "key_export_download_finished"
 
+    const val ANALYTICS_KEY_SHARE_APP = "click_share_app"
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt:11]: These fields don't need ANALYTICS_ prefix, when they are in Analytics class

**Review by davidvavra — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt [hunk 5]
```diff
@@ -203,6 +211,7 @@ class DashboardFragment : BaseFragment<FragmentDashboardPlusBinding, DashboardVM
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return when (item.itemId) {
             R.id.menu_share -> {
+                Analytics.logEvent(requireContext(), ANALYTICS_KEY_SHARE_APP)
```
[the remark this answers is not in the document]
Comment (vandus) [reply app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt:214]: it starts a new activity, cannot be moved after the action. also it's supposed to log the click action, not that a certain screen is opened, so it should not be moved to the screen

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt [hunk 6]
```diff
@@ -61,10 +65,9 @@ class MainActivity :
 
         findNavController(R.id.nav_host_fragment).let {
             bottom_navigation.setOnNavigationItemSelectedListener { item ->
-                navigate(
-                    item.itemId,
-                    navOptions = NavOptions.Builder().setPopUpTo(R.id.nav_graph, false).build()
-                )
+                logTabClickEventToAnalytics(item)
```
[the remark this answers is not in the document]
Comment (vandus) [reply app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt:68]: same as above

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (davidvavra) [reply app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt:214]: Why? The old activity can still execute code, until you call finish()

File: app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt [hunk 7]
```diff
@@ -8,6 +8,17 @@ object Analytics {
...
+    const val ANALYTICS_KEY_PAUSE_APP = "click_pause_app"
+    const val ANALYTICS_KEY_RESUME_APP = "click_resume_app"
+
+    const val ANALYTICS_KEY_HOME = "click_tab_home"
+    const val ANALYTICS_KEY_NEWS = "click_tab_news"
+    const val ANALYTICS_KEY_CONTACTS = "click_tab_contacts"
+    const val ANALYTICS_KEY_HELP = "click_tab_help"
+
+    const val ANALYTICS_KEY_CURRENT_MEASURES = "click_current_measures"
```
Comment (augard) [inline app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt:20]: Klika se mysi, na telefonu mas "tap" ;)

Comment (vandus) [reply app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt:20]: bezne pouzivam tap ale do analytics vzdy vylucne click, asi zvyk :) na androide sa pouziva onClickListener na tapanie, takze je to asi jedno :)

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt [hunk 8]
```diff
@@ -203,6 +211,7 @@ class DashboardFragment : BaseFragment<FragmentDashboardPlusBinding, DashboardVM
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return when (item.itemId) {
             R.id.menu_share -> {
+                Analytics.logEvent(requireContext(), KEY_SHARE_APP)
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt:214]: Log after action

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt [hunk 9]
```diff
@@ -84,6 +87,18 @@ class MainActivity :
         })
     }
 
+    private fun logTabClickEventToAnalytics(item: MenuItem) {
+        L.i("clicked tab ${item.title}")
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/main/MainActivity.kt:91]: Should this log stay?

File: app/src/main/kotlin/cz/covid19cz/erouska/ui/mydata/MyDataFragment.kt [hunk 10]
```diff
@@ -20,22 +21,17 @@ class MyDataFragment :
...
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         if (!AppConfig.updateNewsOnRequest) {
             refresh_container.isEnabled = false
         }
+
+        measures_text.setOnClickListener {
+            Analytics.logEvent(requireContext(), KEY_CURRENT_MEASURES)
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/ui/mydata/MyDataFragment.kt:32]: Log after action

File: app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt [hunk 11]
```diff
@@ -8,6 +8,17 @@ object Analytics {
     const val KEY_EXPORT_DOWNLOAD_STARTED = "key_export_download_started"
     const val KEY_EXPORT_DOWNLOAD_FINISHED = "key_export_download_finished"
 
+    const val KEY_SHARE_APP = "click_share_app"
```
Comment (davidvavra) [inline app/src/main/kotlin/cz/covid19cz/erouska/utils/Analytics.kt:11]: I also have a preference to "tap". OnClickListener is more like a relict from distant past when Android was being born :) But no blocker.

**Review by davidvavra — CHANGES_REQUESTED** [review]

**Review by davidvavra — APPROVED** [review]

**Pull request merged**